module/epochs/epoch_lookup.go

```diff
@@ -39,6 +37,23 @@ func (cache *epochRangeCache) latest() epochRange {
 	return cache[2]
 }
 
+// extendLatestEpoch updates the final view of the latest epoch with the final view of the epoch extension.
+// No errors are expected during normal operation.
+func (cache *epochRangeCache) extendLatestEpoch(extensionFinalView uint64) error {
+	// sanity check: latest epoch should already be cached.
+	if !cache[2].exists() {
+		return fmt.Errorf("sanity check failed: latest epoch does not exist")
+	}
+
```

Comment:
I'd suggest adding a sanity check that `cache[2].counter == epochCounter` (we can pass in `epochCounter` from the `EpochExtended` event, as proposed in a separate comment).

Reply:
https://github.com/onflow/flow-go/pull/6247/commits/95a50bfbea47488bb4c83c547f5cabc16df12473